### PR TITLE
HYDRA-833 - Maya MaterialX material bindings on Maya geometry

### PR DESCRIPTION
**Pull request by vlasovi**
- Support MaterialX on Maya native prims
- Autotest

### CODE REVIEW

**Comment by vlasovi** [comment]
FYI @lanierd-adsk @ppt-adsk @roopavr-adsk @debloip-adsk

File: test/lib/mayaUsd/render/mayaToHydra/testMaterialXOnNative.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        self.assertSnapshotClose(imageName, 
+                                 self.IMAGEDIFF_FAIL_THRESHOLD,
+                                 self.IMAGEDIFF_FAIL_PERCENT)
+
+    @unittest.skipUnless(mtohUtils.checkForPlugin('LookdevXMaya'), "Requires LookDevX Plugin.")
+    def test_MaterialX(self):
+        mayaUtils.openTestScene("testMaterialX", "RedMtlxSphere.ma")
+        self.setBasicCam(2)
+        self.setHdStormRenderer()
+        self.verifySnapshot("RedMtlxSphere.png")
```
Comment (lilike-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testMaterialXOnNative.py:38]: are we missing to pass IMAGEDIFF_FAIL_THRESHOLD and IMAGEDIFF_FAIL_PERCENT here?

File: lib/mayaHydra/hydraExtensions/adapters/materialAdapter.cpp [hunk 2]
```diff
@@ -231,10 +246,81 @@ class MayaHydraShadingEngineAdapter : public MayaHydraMaterialAdapter
...
+        return true;
+    }
+
     VtValue GetMaterialResource() override
     {
         TF_DEBUG(MAYAHYDRALIB_ADAPTER_MATERIALS)
             .Msg("MayaHydraShadingEngineAdapter::GetMaterialResource(): %s\n", GetID().GetText());
+        
+        HdMaterialNetworkMap materialXNetworkMap;
+        if (PopulateMaterialXNetworkMap(materialXNetworkMap)) {
```
Comment (lilike-adsk) [inline lib/mayaHydra/hydraExtensions/adapters/materialAdapter.cpp:320]: is there any reason not to do the translation inside MayaHydraMaterialNetworkConverter? I feel it's better to move the code into MayaHydraMaterialNetworkConverter as MatX is also one kind of Material.

File: test/lib/mayaUsd/render/mayaToHydra/testMaterialXOnNative.py (hunk 1 above)
Comment (vlasovi) [reply test/lib/mayaUsd/render/mayaToHydra/testMaterialXOnNative.py:38]: No, they are passed from inside verifySnapshot method

File: lib/mayaHydra/hydraExtensions/adapters/materialAdapter.cpp (hunk 2 above)
Comment (vlasovi) [reply lib/mayaHydra/hydraExtensions/adapters/materialAdapter.cpp:320]: The interface of MayaHydraMaterialNetworkConverter seems to be developped specifically for UsdPreviewSurface network. I think we would have hard times adapting and rewriting it for the needs of MaterialX. For example method HdMaterialNode* GetMaterial would need to change its output type as well as all other public methods have no meaning for MaterialX (like GetPreviewShaderParams). I don't see any reason for readapting  MayaHydraMaterialNetworkConverter for MaterialX

File: test/lib/mayaUsd/render/mayaToHydra/testMaterialXOnNative.py (hunk 1 above)
Comment (lilike-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/testMaterialXOnNative.py:38]: right, sorry, I missed that. :)

File: lib/mayaHydra/hydraExtensions/adapters/materialAdapter.cpp (hunk 2 above)
Comment (lilike-adsk) [reply lib/mayaHydra/hydraExtensions/adapters/materialAdapter.cpp:320]: ok, sounds like having a base materialConverter and diferent sub materialConverters would make the architecture looks better, as the actual conversion is the responsibility of materialConverter, not materialAdapter.

I'm fine with the current code, just trying to make it easier to maintain,  we could also refactor it later when we need more materialConverters in the future.

**Review by lilike-adsk — APPROVED** [review]

**Pull request merged**